File: backend/src/main/java/com/supermarket/erp/module/system/entity/PaymentMethod.java
package com.supermarket.erp.module.system.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("sys_payment_method")
public class PaymentMethod {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long tenantId;

    private String name;

    private String code;

    private String icon;

    private Integer sort;

    private Integer status;

    private BigDecimal balance;

    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableField(fill = FieldFill.INSERT)
    private Long createBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updateBy;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/paymentMethod.ts
import { get, post, put, del } from './request'

export interface PaymentMethod {
  id: number
  name: string
  code: string
  icon: string | null
  sort: number
  status: number
  balance: number
  remark: string | null
  createTime: string
}

export async function getPaymentMethods(): Promise<PaymentMethod[]> {
  return get<PaymentMethod[]>('/payment-methods')
}

export async function createPaymentMethod(data: Partial<PaymentMethod>): Promise<number> {
  return post<number>('/payment-methods', data)
}

export async function updatePaymentMethod(id: number, data: Partial<PaymentMethod>): Promise<void> {
  return put<void>(`/payment-methods/${id}`, data)
}

export async function deletePaymentMethod(id: number): Promise<void> {
  return del<void>(`/payment-methods/${id}`)
}

export async function rechargePaymentMethod(id: number, amount: number, remark: string): Promise<void> {
  return put<void>(`/payment-methods/${id}/recharge`, { amount, remark })
}
